src/averageAges.js

```diff
@@ -37,7 +52,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter((person) => {
+    const isWoman = person.sex === 'f';
+
+    if (withChildren) {
+      return isWoman && people.some((one) => person.name === one.mother);
+    }
+
+    return isWoman;
```

Comment:
You can simplify the filter condition by directly returning the expression 'person.sex === 'f' && (withChildren ? people.some((one) => person.name === one.mother) : true)'.